src/uct/ib/mlx5/gdaki/gdaki.c

```diff
@@ -613,18 +620,23 @@ static uct_iface_ops_t uct_rc_gdaki_iface_tl_ops = {
             ucs_empty_function_return_unsupported,
 };
 
-static struct ibv_mr *
-uct_rc_gdaki_reg_mr(const uct_ib_md_t *md, void *address, size_t length)
+static ucs_status_t
+uct_rc_gdaki_reg_mr(const uct_ib_md_t *md, const void *address, size_t length,
+                    struct ibv_mr **mr_p)
 {
+    uct_md_mem_reg_params_t params;
     uct_cuda_copy_md_dmabuf_t dmabuf;
 
-    dmabuf = uct_rc_gdaki_get_dmabuf(md, address, length);
-    if (dmabuf.fd == UCT_DMABUF_FD_INVALID) {
-        return ibv_reg_mr(md->pd, address, length, UCT_IB_MEM_ACCESS_FLAGS);
-    }
+    params.field_mask    = UCT_MD_MEM_REG_FIELD_FLAGS |
+                           UCT_MD_MEM_REG_FIELD_DMABUF_FD |
+                           UCT_MD_MEM_REG_FIELD_DMABUF_OFFSET;
+    params.flags         = 0;
+    dmabuf               = uct_rc_gdaki_get_dmabuf(md, address, length);
+    params.dmabuf_fd     = dmabuf.fd;
+    params.dmabuf_offset = dmabuf.offset;
 
-    return ibv_reg_dmabuf_mr(md->pd, dmabuf.offset, length, (uint64_t)address,
-                             dmabuf.fd, UCT_IB_MEM_ACCESS_FLAGS);
+    return uct_ib_reg_mr((uct_ib_md_t*)md, (void*)address, length, &params,
```

Comment:
why need to cast to uct_ib_md_t*)?
also, maybe change address param to be non-const to avoid the (void*) cast?